xptifw/CMakeLists.txt

```diff
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.8)
 set(XPTI_VERSION 0.4.1)
 
 project (xptifw VERSION "${XPTI_VERSION}" LANGUAGES CXX)
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Why do we need C++17? As far as I remember we support build in C++14 mode with disabled C++17 features.
Which features will be missed if C++14 standard will be kept?

Reply:
`std::optional`, `std::string_view`, shared locks, maybe something else I missed.

Note, that we require GCC 7.5.0, which already supports out of the box, and we compile libsycl.so with C++17 enabled by default. We only require C++ 14 in SYCL headers, and even then we only promise limited SYCL 1.2.1 functionality.